tests/mock_server_docker.py

```diff
@@ -96,10 +95,27 @@ def assert_server_process_start_called_with(self,
 
         self._assert_docker_exec_run_with_args(cmd=cmd, stream=True)
 
+    def raise_exception_on_container_run(self):
+        """
+        Raises MockAPIError on container run
+        """
+        self.exception_on_run_patcher = patch(
```

Comment:
This should be patched as
```
self.patcher_api_error = patch(
            'model_analyzer.triton.server.server_docker.docker.errors.APIError',
            MockApiError)
self.api_error_mock = self.patcher_api_error.start()
```
This is so that the `catch` is able to catch the mock. Then this line 

```
self.api_error_mock.from_env.return_value.containers.run.side_effect = self.api_error_mock
```
will actually trigger the catch.